This router has BCM4709A0 SoC, 128 MiB NAND flash (MX30LF1G18AC-TI),
256 MiB memory and 4 x LAN and 1 x WAN ports.
WiFi chips are BCM4360 x 2.
USB2.0 x 1. USB3.0 x 1.
this router cpu port only used port 8,but b53 must set port 7, is it a bug?
see 700-b53-add-hacky-CPU-port-fixes-for-devices-not-using-p.patch


--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -105,6 +105,8 @@ dtb-$(CONFIG_ARCH_BCM_5301X) += \
 	bcm4709-netgear-r7000.dtb \
 	bcm4709-netgear-r8000.dtb \
 	bcm4709-tplink-archer-c9-v1.dtb \
+	bcm4709-arris-sbr-ac1900p.dtb \
+	bcm4709-arris-sbr-ac3200p.dtb \
 	bcm47094-dlink-dir-885l.dtb \
 	bcm47094-linksys-panamera.dtb \
 	bcm47094-luxul-abr-4500.dtb \
--- /dev/null
+++ b/arch/arm/boot/dts/bcm4709-arris-sbr-ac1900p.dts
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/*
+ * DTS for ARRIS SBR-AC1900P
+ *
+ * Copyright (C) 2019 Shi Zhangqiang <lostlonger.g@gmail.com>
+ * Copyright (C) 2019 Xu Zhaowei <paldier@hotmail.com>
+ */
+
+/dts-v1/;
+
+#include "bcm4709.dtsi"
+#include "bcm5301x-nand-cs0-bch4.dtsi"
+
+/ {
+	compatible = "arris,sbr-ac1900p", "brcm,bcm4709", "brcm,bcm4708";
+	model = "ARRIS SBR-AC1900P";
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	memory {
+		reg = <0x00000000 0x08000000
+		       0x88000000 0x08000000>;
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		restart {
+			label = "Reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&chipcommon 19 GPIO_ACTIVE_LOW>;
+		};
+		
+		wps {
+			label = "WPS";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&chipcommon 20 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&uart1 {
+	status = "okay";
+};
+
+&usb2 {
+	vcc-gpio = <&chipcommon 18 GPIO_ACTIVE_HIGH>;
+};
+
+&usb3 {
+	vcc-gpio = <&chipcommon 21 GPIO_ACTIVE_HIGH>;
+};
+
+&usb3_phy {
+	status = "okay";
+};
+
+&nandcs {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "boot";
+			reg = <0x0000000 0x0080000>;
+			read-only;
+		};
+
+		partition@80000 {
+			label = "nvram";
+			reg = <0x0080000 0x0180000>;
+		};
+
+		partition@200000 {
+			label = "firmware";
+			reg = <0x0200000 0x6600000>;
+			compatible = "brcm,trx";
+		};
+
+		partition@6800000 {
+			label = "env";
+			reg = <0x6800000 0x0200000>;
+			read-only;
+		};
+
+		partition@6a00000 {
+			label = "data";
+			reg = <0x6a00000 0x1600000>;
+		};
+	};
+};
+
